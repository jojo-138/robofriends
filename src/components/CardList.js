import { Card } from './Card';

export const CardList = ({ users,input }) => {
    const filteredUser = users.filter(user => user.name.toLowerCase().includes(input.toLowerCase()));
    const user = filteredUser.map(user => {
        return (
            <Card key={user.id} id={user.id} name={user.name} email={user.email}/>
        );
    });
    return (
        <div className="container bn pa3 f7 ma3 vh-75 overflow-y-auto">
            {user}
        </div>
    );
};