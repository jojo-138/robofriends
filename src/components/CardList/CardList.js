import Card from '../Card/Card';

const CardList = ({ users,input }) => {
    const filteredUser = users.filter(user => user.name.toLowerCase().includes(input.toLowerCase()));
    const user = filteredUser.map(user => {
        return (
            <Card key={user.id} id={user.id} name={user.name} email={user.email}/>
        );
    });
    return (
        <div className="container bn pa3 f7 ma3 vh-75 overflow-y-auto" id="card-list">
            {user}
        </div>
    );
}

export default CardList;