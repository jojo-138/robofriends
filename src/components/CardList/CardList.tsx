import { useEffect } from 'react';
import { selectStatus, selectUsers } from '../../features/fetchRobots/robotsSlice';
import { selectInput } from '../../features/setSearchField/searchField';
import { useAppDispatch, useAppSelector } from '../../hooks';
import { fetchRobots } from '../../features/fetchRobots/robotsSlice';
import Card from '../Card/Card';

const CardList = () => {
    const users = useAppSelector(selectUsers);
    const status = useAppSelector(selectStatus);
    const input = useAppSelector(selectInput);
    const dispatch = useAppDispatch();

    useEffect (() => {
        dispatch(fetchRobots());
    }, [dispatch]);
    
    const filteredUser = users.filter(user => user.name.toLowerCase().includes(input.toLowerCase()));

    const user = filteredUser.map(user => {
        return (
            <Card key={user.id} id={user.id} name={user.name} email={user.email}/>
        );
    });

    return (
        <div className="container bn pa3 f7 ma3 vh-75 overflow-y-auto" data-testid="card-list">
            {
                status === 'loading' ? 
                <>Fetching robofriends...</>
                :
                user
            }
        </div>
    );
};

export default CardList;
