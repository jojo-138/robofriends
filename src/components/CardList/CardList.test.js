import { render, screen } from "@testing-library/react";
import CardList from "./CardList";
import ShallowRenderer from 'react-test-renderer/shallow';

describe('CardList Component', () => {
  const mockUsers = [
    {
      id: 1,
      name: 'john',
      email: 'john@gmail.com'
    },
    {
      id: 2,
      name: 'joana',
      email: 'joana@gmail.com'
    }
  ];
  test('renders correctly', () => {
    const renderer = new ShallowRenderer();
    expect(renderer.render(<CardList users={mockUsers} input='' />)).toMatchSnapshot();
  })

  test('filters users correctly', () => {
    render(<CardList users={mockUsers} input='joa' />)
    const name = screen.getByRole('heading');
    expect(name).toHaveTextContent(/joana/i);
    expect(name).not.toHaveTextContent(/john/i);
  })
})