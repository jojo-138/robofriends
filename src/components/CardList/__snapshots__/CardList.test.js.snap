// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList Component renders correctly 1`] = `
<div
  className="container bn pa3 f7 ma3 vh-75 overflow-y-auto"
  id="card-list"
>
  <Card
    email="john@gmail.com"
    id={1}
    name="john"
  />
  <Card
    email="joana@gmail.com"
    id={2}
    name="joana"
  />
</div>
`;
